esphome/components/lvgl/__init__.py

```diff
@@ -0,0 +1,234 @@
+            "1" if config[df.CONF_BYTE_ORDER] == "big_endian" else "0",
+        )
+    add_define(
+        "LV_COLOR_CHROMA_KEY",
+        await lvalid.lv_color.process(config[df.CONF_TRANSPARENCY_KEY]),
+    )
+    CORE.add_build_flag("-Isrc")
+
+    cg.add_global(lvgl_ns.using)
+    lv_component = cg.new_Pvariable(config[CONF_ID])
+    await cg.register_component(lv_component, config)
+    Widget.create(config[CONF_ID], lv_component, WIDGET_TYPES[df.CONF_OBJ], config)
+    displays = get_display_list(config)
+    for display in displays:
+        cg.add(lv_component.add_display(await cg.get_variable(display)))
+
+    frac = config[CONF_BUFFER_SIZE]
+    if frac >= 0.75:
+        frac = 1
+    elif frac >= 0.375:
+        frac = 2
+    elif frac > 0.19:
+        frac = 4
+    else:
+        frac = 8
+    cg.add(lv_component.set_buffer_frac(int(frac)))
+    cg.add(lv_component.set_full_refresh(config[df.CONF_FULL_REFRESH]))
+
+    for font in helpers.esphome_fonts_used:
+        await cg.get_variable(font)
+        cg.new_Pvariable(ID(f"{font}_engine", True, type=FontEngine), MockObj(font))
+    default_font = config[df.CONF_DEFAULT_FONT]
+    if default_font not in helpers.lv_fonts_used:
+        add_define(
+            "LV_FONT_CUSTOM_DECLARE", f"LV_FONT_DECLARE(*{df.DEFAULT_ESPHOME_FONT})"
+        )
+        globfont_id = ID(
+            df.DEFAULT_ESPHOME_FONT,
+            True,
+            type=lv_font_t.operator("ptr").operator("const"),
+        )
+        cg.new_variable(globfont_id, MockObj(default_font))
+        add_define("LV_FONT_DEFAULT", df.DEFAULT_ESPHOME_FONT)
+    else:
+        add_define("LV_FONT_DEFAULT", default_font)
+
+    with LvContext():
+        await set_obj_properties(lv_scr_act, config)
+        await add_widgets(lv_scr_act, config)
+    Widget.set_completed()
+    await add_init_lambda(lv_component, LvContext.get_code())
+    for comp in helpers.lvgl_components_required:
+        CORE.add_define(f"USE_LVGL_{comp.upper()}")
+    for use in helpers.lv_uses:
+        add_define(f"LV_USE_{use.upper()}")
+    lv_conf_h_file = CORE.relative_src_path(LV_CONF_FILENAME)
+    write_file_if_changed(lv_conf_h_file, generate_lv_conf_h())
+    CORE.add_build_flag("-DLV_CONF_H=1")
+    CORE.add_build_flag(f'-DLV_CONF_PATH="{LV_CONF_FILENAME}"')
+
+
+FINAL_VALIDATE_SCHEMA = final_validation
+
+CONFIG_SCHEMA = (
+    cv.polling_component_schema("1s")
+    .extend(obj_schema("obj"))
+    .extend(
+        {
+            cv.Optional(CONF_ID, default=df.CONF_LVGL_COMPONENT): cv.declare_id(
+                LvglComponent
+            ),
+            cv.GenerateID(df.CONF_DISPLAYS): cv.Any(
+                cv.use_id(Display),
+                cv.ensure_list(
+                    cv.maybe_simple_value(
+                        {
+                            cv.Required(CONF_DISPLAY_ID): cv.use_id(Display),
+                        },
+                        key=CONF_DISPLAY_ID,
+                    ),
+                ),
```

Comment:
Does multi display really need a `display_id` key for each one?
A simple list of ids can work too (goes with my above suggestion in `get_display_list`)
```suggestion
                cv.use_id(Display), cv.ensure_list(cv.use_id(Display))
```